opensearch-service-domain-cdk/test/stack-composer.test.ts

```diff
@@ -0,0 +1,225 @@
+import {App} from "aws-cdk-lib";
+import {StackComposer} from "../lib/stack-composer";
+import {Template} from "aws-cdk-lib/assertions";
+
+test('Test missing domain name throws error', () => {
+
+    const app = new App({
+        context: {
+            domainName: ""
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+       env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test missing engine version throws error', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: ""
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test invalid engine version format throws error', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: "OpenSearch_1.3"
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test ES engine version format is parsed', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: "ES_7.10"
+        }
+    })
+
+    const openSearchStacks =  new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    const domainStack = openSearchStacks.stacks.filter((s) => s.stackName === "opensearchDomainStack")[0]
+    const domainTemplate = Template.fromStack(domainStack)
+    domainTemplate.resourceCountIs("AWS::OpenSearchService::Domain", 1)
+})
+
+test('Test OS engine version format is parsed', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: "OS_1.3"
+        }
+    })
+
+    const openSearchStacks =  new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    const domainStack = openSearchStacks.stacks.filter((s) => s.stackName === "opensearchDomainStack")[0]
+    const domainTemplate = Template.fromStack(domainStack)
+    domainTemplate.resourceCountIs("AWS::OpenSearchService::Domain", 1)
```

Comment:
I might be missing something, but how does this test verify that the engineVersion is parsed? Is it just a smoke test (no error is thrown)? (same question for the ES test above)

Reply:
Yes this currently is more of a smoke test to make sure it is accepted. This should probably be enhanced to actually check the version of the domain resource, I think I was in the mindset of not wanting to make any assumptions about the CFN structure but I had to make this tradeoff to make some of the other tests valuable